tests/unit/s2n_hkdf_test.c

```diff
@@ -0,0 +1,339 @@
+/*
+ * Copyright 2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+
+#include <string.h>
+
+#include "testlib/s2n_testlib.h"
+#include "stuffer/s2n_stuffer.h"
+#include "crypto/s2n_hmac.h"
+#include "crypto/s2n_hkdf.h"
+#include "utils/s2n_blob.h"
+
+#define NUM_5869_TESTS 7
+#define MAX_OUT_KEY_SIZE 82
+#define MAX_PSEUDO_RAND_KEY_SIZE 32
+
+struct hkdf_test_vector {
+    s2n_hmac_algorithm alg;
+    uint8_t in_key[80];
+    uint32_t in_key_len;
+    uint8_t salt[80];
+    uint32_t salt_len;
+    uint8_t info[80];
+    uint32_t info_len;
+    uint8_t pseudo_rand_key[MAX_PSEUDO_RAND_KEY_SIZE];
+    uint32_t prk_len;
+    uint8_t out_key[MAX_OUT_KEY_SIZE];
+    uint32_t out_key_len;
+};
+
+/*
+ * Test vectors obtained from RFC 5869:
+ * Includes SHA-256 and SHA-1 vectors
```

Comment:
Well done those RFC authors! awesome to have good vectors like this in the test.